pkg/backup/table_pattern.go

```diff
@@ -109,6 +111,58 @@ func getTableListByPatternLocal(metadataPath string, tablePattern string, skipTa
 	return result, nil
 }
 
+func getTableListByRestoreDatabaseMappingRule(originTables *ListOfTables, dbMapRule map[string]string) error {
+	result := ListOfTables{}
+	dbMetaMap := make(map[string]metadata.TableMetadata, 0)
+	for i := 0; i < len(*originTables); i++ {
+		table := (*originTables)[i]
+		dbMetaMap[table.Database] = table
+	}
+	for i := 0; i < len(*originTables); i++ {
+		originDBMeta := (*originTables)[i]
+		if targetDB, ok := dbMapRule[originDBMeta.Database]; ok {
+			// substitute database in the table create query
+			regExp := regexp.MustCompile(`(?m)^(CREATE|ATTACH) (TABLE|VIEW|MATERIALIZED VIEW|DICTiONARY|FUNCTION) ([\x60]?)([^\s\x60\.]*)([\x60]?)\.([^\s\x60\.]*)(?:( TO )([\x60]?)([^\s\x60\.]*)([\x60]?)(\.))?`)
+			var substitution string
+
+			if len(regexp.MustCompile(`(?m)^CREATE`).FindAllString(originDBMeta.Query, -1)) > 0 {
+				// matching CREATE... command
+				substitution = fmt.Sprintf("${1} ${2} ${3}%v${5}.${6}", targetDB)
+			} else if len(regexp.MustCompile(`(?m)^ATTACH`).FindAllString(originDBMeta.Query, -1)) > 0 {
+				// matching ATTACH...TO... command
+				substitution = fmt.Sprintf("${1} ${2} ${3}%v${5}.${6}${7}${8}%v${11}", targetDB, targetDB)
+			} else {
+				return fmt.Errorf("there is an error when substituting the default database in query, try to open an issue on Github and we will fix it :)")
+			}
+
+			originDBMeta.Query = regExp.ReplaceAllString(originDBMeta.Query, substitution)
+			originDBMeta.Database = targetDB
+			result = append(result, originDBMeta)
+		}
+		// FIXME: Need to optimize
```

Comment:
Could you clarify, what exactly this part of code do?

`table.Parts` doesn't split by `database` it splitted by diskname